djaddon/git.py

```diff
@@ -9,9 +9,10 @@
 def _log_git_status(func):
     @wraps(func)
     def with_logging(*args, **kwargs):
-        ret = func(*args, **kwargs)
-        for key in (args[0] - args[0].GitKey()).project().fetch.as_dict:
-            args[0].GitKey().log_key(key)
+        with dj.conn().transaction:
```

Comment:
Explicit calling of `dj.conn()` here is problematic - in the (near) future, we want to support multiple connections, thus we need to ensure that we minimize the reference to global connection fetcher like `dj.conn()` call. I believe that the first argument to the `with_logging` is `self`, so we should replace the call to `dj.conn()` with `self.connection`